### ISSUE TITLE
Upgrade to 2.8.0-rc2 does not add dsahostkeypub field to cluster config

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#616]
Originally reported of Google Code with ID 560.

```
<b>What steps will reproduce the problem?</b>
1. Create a cluster with 2.8.0-rc1 (and possibly earlier)
2. Upgrade to 2.8.0-rc2
3. Luxid fails to respond to queries (e.g. gnt-cluster info), saying:

ganeti-luxid pid=26957 ERROR Failed to load config: parsing configuration: key 'cluster': key 'dsahostkeypub' not found, object contains only ["mac_prefix","serial_no","hvparams","diskparams","uid_pool","prealloc_wipe_disks","os_hvp","osparams","shared_file_storage_dir","master_netmask","uuid","use_external_mip_script","default_iallocator","master_ip","cluster_name","mtime","nicparams","highest_used_port","enabled_disk_templates","maintain_node_health","modify_ssh_setup","tags","ipolicy","tcpudp_port_pool","candidate_pool_size","file_storage_dir","blacklisted_os","enabled_hypervisors","drbd_usermode_helper","reserved_lvs","ctime","modify_etc_hosts","hidden_os","master_netdev","rsahostkeypub","ndparams","master_node","primary_ip_family","volume_group_name","beparams"]

Commit a9542a4feda0 added support for DSA host keys, including an extra field in the cluster configuration, but didn't handle updates (using e.g. config.ConfigWriter._UpgradeConfig())
```
Originally added on 2013-08-30 10:54:13 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Perhaps the field should just be optional in the config?

Thanks,

Guido
```
Originally added on 2013-08-30 11:00:07 +0000 UTC.

Added Labels: Priority-High Type-Defect
Added to Milestone: Release2.8

**Comment by Ganeti-Issues-Migrator** [comment]
```
Well, IMHO it doesn't make much sense, there's almost always a DSA host 
key available.

What would perhaps make more sense, would be to create a single field, 
"hostkeys", as an array of (type, key) tuples. This would also be more
future-proof, as there are already e.g. ECDSA keys which should ideally 
be handled as well.
```
Originally added on 2013-08-30 11:09:40 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Isn't it a bit late to do that in 2.8? We should go for the minimum fix needed, and for a field that wasn't there before and that can be empty (esp. on old clusters) perhaps this should be empty. In master we can move both fields to a common one and talk about how to handle different keys. Don't you agree?
```
Originally added on 2013-08-30 11:13:54 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I wasn't thinking of 2.8 but rather long-term. Yes, a minimal 
backwards-compatible fix should do. Also downgrade to 2.7 should remove 
the field.
```
Originally added on 2013-08-30 11:16:41 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This was fixed for 2.8 with the following commits:

commit 78237f71007b1cde1bbd06b23a0761d211e9b98a
Author: Apollon Oikonomopoulos <apoikos@gmail.com>
Date:   Sun Sep 1 23:37:08 2013 +0300

    Remove dsahostkeypub during config downgrade
    
    Remove the newly-added DSA host public key field from the configuration file
    when downgrading to 2.7, if present. Otherwise masterd will refuse to start.
    
    Signed-off-by: Apollon Oikonomopoulos <apoikos@gmail.com>
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

commit 3022ea9f332cb5835023987d13070742199a0a12
Author: Apollon Oikonomopoulos <apoikos@gmail.com>
Date:   Sun Sep 1 23:36:29 2013 +0300

    Mark the DSA host pubkey as optional
    
    Commit a9542a4 introduced support for DSA SSH keys. However, the dsahostkeypub
    field added to the config is not marked as optional in the Haskell components.
    As a result, luxid thinks the config file is corrupt and refuses to start. We
    fix this by marking the dsahostkeypub as an optionalField.
    
    This fixes issue 560.
    
    Signed-off-by: Apollon Oikonomopoulos <apoikos@gmail.com>
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>
```
Originally added on 2013-09-11 07:57:38 +0000 UTC.

Mentioned Issue with GoogleCode ID 560 has been migrated to GitHub with ID #616.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-10-30 09:30:16 +0000 UTC.

Changed State: Released
